src/dht.cpp

```diff
@@ -2120,6 +2120,42 @@ Dht::bucketMaintenance(RoutingTable& list)
 }
 
 void
+Dht::maintainStorage(InfoHash id) {
+    auto *local_storage = findStorage(id);
+    std::vector<std::shared_ptr<Value>> foreign_values;
+
+    if (!local_storage) { return; }
+    get(id,
+    [&foreign_values](std::shared_ptr<Value> value){
+        foreign_values.push_back(value);
+        return true;
+    },
+    [=,&foreign_values](bool success, const std::vector<std::shared_ptr<Node>> &nodes) {
+        if (success) {
+            auto this_node = std::find_if(nodes.begin(), nodes.end(), [this](const std::shared_ptr<Node>& node) {
+                        return getNodeId() == node->id;
+                    });
+            //If this node is not in the list of 8 nodes nearby the given id.
```

Comment:
this node will never be in the list (our own node is not part of routing tables). Instead we should check if we are XOR-closer to "id" than the XOR-further node in the list.